sdk/src/types/block/semantic.rs

```diff
@@ -17,91 +17,110 @@ use crate::types::block::{
 
 /// Errors related to ledger types.
 #[derive(Debug)]
-pub enum ConflictError {
-    /// Invalid conflict byte.
-    InvalidConflict(u8),
+pub enum TransactionFailureError {
+    /// Invalid transaction failure reason byte.
+    InvalidReason(u8),
 }
 
-impl fmt::Display for ConflictError {
+impl fmt::Display for TransactionFailureError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Self::InvalidConflict(byte) => write!(f, "invalid conflict byte {byte}"),
+            Self::InvalidReason(byte) => write!(f, "invalid transaction failure reason byte {byte}"),
         }
     }
 }
 
-impl From<Infallible> for ConflictError {
+impl From<Infallible> for TransactionFailureError {
     fn from(err: Infallible) -> Self {
         match err {}
     }
 }
 
 #[cfg(feature = "std")]
-impl std::error::Error for ConflictError {}
+impl std::error::Error for TransactionFailureError {}
 
-/// Represents the different reasons why a transaction can conflict with the ledger state.
+/// Describes the reason of a transaction failure.
 #[repr(u8)]
 #[derive(Debug, Copy, Clone, Eq, PartialEq, packable::Packable)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-#[packable(unpack_error = ConflictError)]
-#[packable(tag_type = u8, with_error = ConflictError::InvalidConflict)]
-pub enum ConflictReason {
-    /// The block has no conflict.
-    None = 0,
-    /// The referenced Utxo was already spent.
+#[packable(unpack_error = TransactionFailureError)]
+#[packable(tag_type = u8, with_error = TransactionFailureError::InvalidReason)]
+pub enum TransactionFailureReason {
+    /// The referenced UTXO was already spent.
     InputUtxoAlreadySpent = 1,
-    /// The referenced Utxo was already spent while confirming this milestone.
-    InputUtxoAlreadySpentInThisMilestone = 2,
-    /// The referenced Utxo cannot be found.
-    InputUtxoNotFound = 3,
+    /// The transaction is conflicting with another transaction. Conflicting specifically means a double spend
+    /// situation that both transaction pass all validation rules, eventually losing one(s) should have this reason.
```

Comment:
Isn't `InputUtxoAlreadySpent` also double spend ?

Reply:
Ask in the TIP :trollface: